src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php

```diff
@@ -1250,4 +1227,38 @@ protected static function getMutatorMethods($class)
 
         return $matches[1];
     }
+
+    /**
+     * Get the plain value (no relation) of a model respecting the models mutator, casts and dates property.
+     *
+     * @param  string  $key
+     * @param  mixed $value
+     * @return mixed|array
+     */
+    private function getValue($key, $value)
```

Comment:
This shouldn't be `private`; the framework never(*) uses this visibility as it prevents customizations.

This should have been `protected`.

(*) yeah, they're like 10 references in the whole codebase

Reply:
@mfn is there a guideline / doc where this (no use of privates) is defined? Maybe with other guidelines to follow?

Reply:
I'm not aware.

The Frameworks thousands like of code are the guidelines I guess.

I've had the pleasure to customize a lot of Laravel internals and that's why I know there are almost no private methods compared to that everything else is protected; which is the reason it's so easy to extend.